spinalcordtoolbox/reports/slice.py

```diff
@@ -444,3 +456,44 @@ def get_center(self, img_idx=-1):
         size_y = self.axial_dim(image)
         size_x = self.coronal_dim(image)
         return np.ones(dim) * size_x / 2, np.ones(dim) * size_y / 2
+
+    def mosaic(self, nb_column=0, dilation=(0, 0, 0)):
+        """Obtain matrices of the mosaics
+
+        Mosaic images are cropped based on the bounding box of the spinal cord segmentation.
+
+        :param nb_column: number of mosaic columns
+        :param dilation: a tuple containing (x,y,z) integers for dilating the bounding box used for cropping.
+        :return: tuple of numpy.ndarray containing the mosaics of each slice pixels
+        """
+        # 0. Use the segmentation image to initialize the image cropper
+        self._image_seg = self._images[-1]
+        cropper = ImageCropper()
+        cropper.get_bbox_from_mask(self._image_seg)
```

Comment:
By using the segmentation image to get the bounding box, and cropping the images based on this bounding box, I believe I now have a working sagittal QC:

![image](https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/16181459/7fb547b7-2211-46b1-b540-125a666fd6bc)

(The lesion segmentation toggle works exactly as expected, too!)

Reply:
The command I used to generate this image was exactly the same command in the PR description, just with `-plane Sagittal` instead of `-plane Axial`:

```
-i dcm-zurich-lesions/sub-05/anat/sub-05_acq-ax_T2w.nii.gz -p sct_deepseg_lesion -plane Sagittal -d dcm-zurich-lesions/derivatives/labels/sub-05/anat/sub-05_acq-ax_T2w_label-lesion.nii.gz -s dcm-zurich-lesions/derivatives/labels/sub-05/anat/sub-05_acq-ax_T2w_label-SC_mask-manual.nii.gz -qc ./dcm-zurich-lesions-qc
```

Reply:
This is amazing! Just tested and working fine!
Just one small thing, we could add `A`, `P`, etc. labels to the first sagittal image. Similarly, as is done for axial QC: 
<img width="99" alt="image" src="https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/39456460/45704cfd-fb28-4835-9222-1ad0d2521e6f">

Reply:
I would love to add these labels! Looking into it, though, the current implementation makes this really tricky. :sweat:

For Axial mosaics, the `_add_orientation_label` method assumes a square size of 30x30, and so it _hardcodes the label locations_ (yikes!):

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/a058fc2937cdf9a36a555e79af4420dc52f9e3a7/spinalcordtoolbox/reports/qc.py#L423-L426

To be able to generalize this to Sagittal images of varying shapes, we would need to have access to the bounding box dimensions. But... we lose that context once we enter the `_add_orientation_label` method, which is called way after generating the mosaic image.

So, I think in order to properly position the orientation labels, we would need to move the orientation label step inside of the `mosaic` method, which would require a bigger refactor. :sweat:

Reply:
Ok, after trying to get the information about label placement from `Sagittal.mosaic` to `QcImage._add_orientation_label`, I think adding orientation labels is a complicated enough change that we should do it in a followup PR, possibly after a cleanup PR as well.